dags/general/monitoring.py

```diff
@@ -0,0 +1,133 @@
+from airflow import DAG
+import json
+from operators.mattermost_operator import MattermostOperator
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.http_operator import SimpleHttpOperator
+from datetime import datetime
+from airflow_utils import send_alert
+from airflow.models.xcom import XCom
+from airflow.utils.db import provide_session
+from airflow.models import Variable
+import logging
+task_logger = logging.getLogger('airflow.task')
+
+
+"""
+This method returns list of extractions that are failing
+tap_exit_status = 1, failing
+tap_exit_status = 0, success
+tap_exit_status = None, extraction in process
+"""
+def stitch_check_extractions(response):
+    task_logger.info('Got extractions response from stitch api, checking for errors')
+    failed_extractions={}
+    try:
+        extractions = json.loads(response)['data']
+        for extraction in extractions:
+            if extraction['tap_exit_status'] is 1:
+                failed_extractions[extraction['source_id']] = extraction['tap_description']
+    except Exception as e:
+        task_logger.error('error in getting extractions from stitch ', str(e))
+        raise(e)
+    return failed_extractions
+
+
+"""
+This method returns object of loads that are failing
+error_state != None, failing
+error_state = None, success
+"""
+def stitch_check_loads(response):
+    task_logger.info('Got loads response from stitch api, checking for errors')
+    failed_loads={}
+    try:
+        loads = json.loads(response)['data']
+        for load in loads:
+            if load['error_state'] is not None:
+                failed_loads[load['source_name']] = load['error_state']['notification_data']['error']
+    except Exception as e:
+        task_logger.error('error in getting loads from stitch ', str(e))
+        raise(e)
+    return failed_loads
+
+"""
+This method fetches failed stitch extractions from xcom
+triggers a mattermost alert in case of failure
+""" 
+def resolve_stitch(**kwargs):
+    ti = kwargs['ti']
```

Comment:
Since this function is expecting `ti` as argument, it's a good idea to define it. Function definition could be:

```
def resolve_stitch(ti=None, **kwargs):
```